server_tests/test/clientTests.js

```diff
@@ -1507,213 +1400,199 @@ describe("RUST API TEST SUITE", function() {
                 if (timeElapsed > maxAcceptableTime) {
                     done(new Error('operation took too long. timeElapsed : ' + timeElapsed + ' > maxAcceptableTime : ' + maxAcceptableTime));
                 } else {
-                    done();
                 }
-            }));
+            }, shouldNotGetHere);
         });
     });
 
     // BUBL CAPTURE VIDEO
     describe("Testing /osc/commands/execute camera._bublCaptureVideo endpoint", function() {
         var sessionId;
 
-        before( function(done) {
+        before( function() {
             if (!isBublcam) {
                 return this.skip();
             }
 
-            testClient.startSession()
-            .then( Comparison.catchExceptions(done, function(res) {
+            return testClient.startSession()
+            .then( function onSuccess (res) {
                 sessionId = res.body.results.sessionId;
                 Comparison.oscSessionOpOutput(res, {'sessionId': sessionId});
                 return Utility.restoreDefaultOptions(defaultOptionsFile);
-            }))
-            .then( Comparison.catchExceptions(done, function(res) {
+            }, shouldNotGetHere)
+            .then( function onSuccess (res) {
                 Comparison.oscSetOptionsOutput(res);
-                done();
-            }));
+            }, shouldNotGetHere);
         });
 
-        afterEach( function(done) {
+        afterEach( function() {
             this.timeout(timeoutValue);
             Utility.restoreDefaultOptions(defaultOptionsFile)
-            .then( Comparison.catchExceptions(done, function(res) {
+            .then( function onSuccess (res) {
                 Comparison.oscSetOptionsOutput(res);
-                done();
-            }));
+            }, shouldNotGetHere);
         });
 
-        after( function(done) {
-            testClient.closeSession(sessionId)
-            .then( Comparison.catchExceptions(done, function(res) {
+        after( function() {
+            return testClient.closeSession(sessionId)
+            .then( function onSuccess (res) {
                 Comparison.oscCloseSessionOutput(res);
-                done();
-            }));
+            }, shouldNotGetHere);
         });
 
-        it("Expect success.  camera._bublCaptureVideo successfully captures a video", function(done) {
+        it("Expect success.  camera._bublCaptureVideo successfully captures a video", function() {
             this.timeout(timeoutValue);
             var stopped = false;
-            testClient.bublCaptureVideo(sessionId, function(res) {
+            return testClient.bublCaptureVideo(sessionId, function(res) {
```

Comment:
Yes, using deferred is necessary when initiating new promises from inside a status callback.